src/test/java/api/loans/LoanAPITests.java

```diff
@@ -1840,6 +1840,79 @@ public void canCreateMultipleLoansWithServicePoints()
 
   }
 
+  @Test
+  public void canGetLoanWithPatronGroupName() throws InterruptedException, MalformedURLException, TimeoutException, ExecutionException {
+    IndividualResource staffPatronGroup = patronGroupsFixture.staff();
+
+    LoanBuilder storageLoanBuilder = new LoanBuilder()
+      .withId(UUID.fromString("c56395a7-59bc-4fda-903e-f6bccafd8569"))
+      .withUserId(UUID.fromString("513ffb94-dc62-4c06-8fe4-b6a88d90aef0"))
+      .withItemId(UUID.fromString("6e1c8601-77fb-4feb-97d3-eeb55e5c3950"))
+      .withLoanDate(DateTime.parse("2017-03-06T16:04:43.000+02:00"))
+      .withPatronGroupAtCheckout(staffPatronGroup.getId())
+      .withStatus("Open");
+
+    JsonObject storageLoan = storageLoanBuilder.create();
+
+    loansStorageClient.create(storageLoan);
```

Comment:
To me, the intention behind this change is to remember the patron group during the check out process.

This test feels somewhat artificial to me, in the sense that it demonstrates a workflow that we wouldn't expect clients to make.

Can you help me understand why the scenario is being created like this, rather than as part of checking out an item?

Reply:
At first, while user checking out the item by barcode, the loan will be store in the mod-circulation-storage with "patronGroupIdAtCheckout". Then, when the specific loan will be retrieved from the mod-circulation-storage, the current mechanism will make the call to the /groups by value of the field "patronGroupIdAtCheckout" (which was from the mod-circulation-storage). And then, the response will be constructed to the PatronGroup-object, and gives to the client via LoanRepresentation.

So, the main purpose of this test is: 
Created specific loan (with "patronGroupIdAtCheckout" field) and put it to the storage. Then this loan will be retrieved from the storage and asserts that completed loan has "patronGroupAtCheckout" field with patron`s group name. (In this case - "staff")

Reply:
@VRohach Thank you for explaining your reasoning.

What I don't understand is why fake the context in this test, rather than perform a check out and then a get of the loan?

I think that better demonstrates the workflow that a client is likely to use.

Reply:
@marcjohnson-kint So, you want to remove these 2 test-cases, and only get "patronGroupAtCheckout" field from loans where "checkOutByBarcode" method was called on loansFixture. Am I right?

Reply:
@marcjohnson-kint Yes, I agree with you about removing POST on /circulation/loans, because it adds more complexity for understanding tests in this module.

Reply:
I've raised [CIRC-402](https://issues.folio.org/browse/CIRC-402) for removing the `POST /circulation/loans` endpoint.

Reply:
> @marcjohnson-kint So, you want to remove these 2 test-cases, and only check "patronGroupAtCheckout" field from loans where "checkOutByBarcode" method was called on loansFixture. Am I right?

Yes, I would include the check for whether the patron group information is part of the loan information after a check out.

And when a loan is fetched, either individually or collectively.

I would probably also add a test that demonstrates it also being present once the loan is anonymized. 

Unfortunately, because anonymization is performed in storage, that does require the test to manipulate the loan directly in storage. (I checked, there does not appear to be any tests for anonymized loans that could act as an example, please let me know If I can help to set that up)

Reply:
@marcjohnson-kint I added new tests for checking on this object after anonymize. Could you please review it?